src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/catke_vertical_diffusivity.jl

```diff
@@ -263,28 +262,34 @@ end
     ℓu = momentum_mixing_lengthᶜᶜᶠ(i, j, k, grid, closure, velocities, tracers, buoyancy, surface_buoyancy_flux)
     κu = ℓu * w★
     κu_max = closure.maximum_viscosity
-    return min(κu, κu_max)
+    κu★ = min(κu, κu_max)
+    FT = eltype(grid)
+    return convert(FT, κu★)
 end
```

Comment:
For this PR I suggest changing this to a a type annotation (`κu★::FT`) since this is what we will want in the long run

Reply:
Unfortunately that does not work, but maybe I can pinpoint where the conversion is happening then we won't need any conversion here.

Reply:
Sorry I am not suggesting this as a solution, but rather as a way to catch a bug in the future.

Reply:
Ah I don't think I actually know how type annotations work. So if we say
```julia
return κu★::FT
```
and `κu★` is not of type `FT` then an error/exception will be thrown?

Reply:
I think so, isn't that what you found?

Reply:
It's a bug checking mechanism which may be more broadly useful as we try to prevent spurious promotion

Reply:
No idea why I thought it was a different mechanism to convert types lol. But yes that is what I found.